packages/protocol/contracts/L1/libs/LibAuction.sol

```diff
@@ -0,0 +1,268 @@
+    {
+        return uint64((blockId - 1) / config.auctionBatchSize);
+    }
+
+    // According to David's suggestion:
+    // https://github.com/taikoxyz/taiko-mono/pull/13831#discussion_r1211886142
+    function isPreviousAuctionEnded(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint256 batchId
+    )
+        internal
+        view
+        returns (bool result)
+    {
+        if(batchId == 0) {
+            return true;
+        }
+
+        //Otherwise batchId is above 0, so we can deduct 1 safely
+        TaikoData.Auction memory auction = state.auctions[(batchId-1) % config.auctionRingBufferSize];
+        if(auction.startedAt != 0 && block.timestamp > auction.startedAt + config.auctionWindowInSec) {
+            return true;
+        }
+    }
+
+    // isBidAcceptable determines is checking if the bid is acceptable based on
+    // the defined
+    // criteria. Shall be called after isBatchAuctionable() returns true.
+    function isBidAcceptable(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        TaikoData.Bid calldata newBid
+    )
+        internal
+        view
+        returns (bool result)
+    {
+        TaikoData.Bid memory winningBid = state.auctions[newBid.batchId].bid;
+
+        if (
+            newBid.feePerGas >= config.auctionSmallestGasPerBlockBid
+                && newBid.feePerGas
+                    <= (
+                        (
+                            winningBid.feePerGas
+                                - ((winningBid.feePerGas * 9000) / 10_000)
+                        )
+                    ) // 90%
+                && newBid.deposit
+                    <= ((winningBid.deposit - ((winningBid.deposit * 5000) / 10_000))) // 50%
+                // Commenting out this one, because we need a scoring /
+                    // weighting system rather then just checking if
+                // committedProofWindow is smaller. ProofWindow might be bigger
+                    // but what if the feePerGas is much better.. ?
+                // &&
+                // newBid.committedProofWindow <=
+                    // winningBid.committedProofWindow
+        ) {
+            result = true;
+        }
+    }
+
+    // isBatchAuctionable determines whether a new bid for a batch of blocks
+    // would be accepted or not. 'open ended' - so returns true if no bids came
+    // yet
+    function isBatchAuctionable(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint256 batchId
+    )
+        internal
+        view
+        returns (bool result)
+    {
+        // We should expose this function so that clients could query.
+        // We also need to be sure, that the batchId of bid is indeed valid
+        if (batchId != blockIdToBatchId(config, batchId)) {
+            revert L1_ID_NOT_BATCH_ID();
+        }
+
+        if (!isPreviousAuctionEnded(state, config, batchId)){
```

Comment:
This batch can be auctioned if the previous batch's action has started (not ended)